pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/event_handler.go

```diff
@@ -0,0 +1,162 @@
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/openyurtio/openyurt/cmd/yurt-manager/names"
+	"github.com/openyurtio/openyurt/pkg/apis/apps/v1beta1"
+	"github.com/openyurtio/openyurt/pkg/apis/network"
+	"github.com/openyurtio/openyurt/pkg/apis/network/v1alpha1"
+)
+
+func NewPoolServiceEventHandler() handler.EventHandler {
+	return handler.Funcs{
+		CreateFunc: func(event event.CreateEvent, limitingInterface workqueue.RateLimitingInterface) {
+			handlePoolServiceNormal(event.Object, limitingInterface)
+		},
+		UpdateFunc: func(updateEvent event.UpdateEvent, limitingInterface workqueue.RateLimitingInterface) {
+			handlePoolServiceUpdate(updateEvent.ObjectOld, updateEvent.ObjectNew, limitingInterface)
+		},
+		DeleteFunc: func(deleteEvent event.DeleteEvent, limitingInterface workqueue.RateLimitingInterface) {
+			handlePoolServiceNormal(deleteEvent.Object, limitingInterface)
+		},
+		GenericFunc: func(genericEvent event.GenericEvent, limitingInterface workqueue.RateLimitingInterface) {
+			handlePoolServiceNormal(genericEvent.Object, limitingInterface)
+		},
+	}
+}
+
+func handlePoolServiceNormal(event client.Object, q workqueue.RateLimitingInterface) {
+	ps := event.(*v1alpha1.PoolService)
+	serviceName := getServiceNameFromPoolService(ps)
+	enqueueService(ps.Namespace, serviceName, q)
+}
+
+func getServiceNameFromPoolService(poolService *v1alpha1.PoolService) string {
+	if poolService.Labels == nil {
+		return ""
+	}
+
+	serviceName, ok := poolService.Labels[network.LabelServiceName]
+	if !ok {
+		return ""
+	}
+	return serviceName
+}
+
+func enqueueService(namespace, serviceName string, q workqueue.RateLimitingInterface) {
+	if len(serviceName) == 0 || len(namespace) == 0 {
+		return
+	}
+
+	q.Add(reconcile.Request{
+		NamespacedName: types.NamespacedName{Namespace: namespace, Name: serviceName},
+	})
+}
+
+func handlePoolServiceUpdate(oldObject, newObject client.Object, q workqueue.RateLimitingInterface) {
+	oldPs := oldObject.(*v1alpha1.PoolService)
+	newPs := newObject.(*v1alpha1.PoolService)
+
+	oldServiceName := getServiceNameFromPoolService(oldPs)
+	newServiceName := getServiceNameFromPoolService(newPs)
+
+	if oldServiceName != newServiceName {
+		klog.Warningf("service name of %s/%s is changed from %s to %s", oldPs.Namespace, oldPs.Name, oldServiceName, newServiceName)
+		enqueueService(oldPs.Namespace, oldServiceName, q)
+		enqueueService(newPs.Namespace, newServiceName, q)
+		return
+	}
+	enqueueService(newPs.Namespace, newServiceName, q)
+
+	return
+}
+
+func NewNodePoolEventHandler(c client.Client) handler.EventHandler {
+	return handler.Funcs{
+		CreateFunc: func(createEvent event.CreateEvent, limitingInterface workqueue.RateLimitingInterface) {
+			allMultiRegionLbServicesEnqueue(c, limitingInterface)
+		},
+		UpdateFunc: func(updateEvent event.UpdateEvent, limitingInterface workqueue.RateLimitingInterface) {
+			handleNodePoolUpdate(c, updateEvent, limitingInterface)
```

Comment:
handleNodePoolUpdate --> allLoadBalancerSetServicesEnqueue, don't need to verify labels change.